Update the text about the response returned by the browser API (#214)
openid/OpenID4VP: openid-4-verifiable-presentations-1_0.md

## Response

> Every OpenID4VP Authorization Request results in a response being provided through the W3C Digital Credentials API. The response is an instance of a `DigitalCredential` credential interface, as defined in [@!w3c.digital_credentials_api], and the OpenID4VP Authorization Response parameters as defined for the Response Type are represented as an object within the `data` attribute.

Reviewer: ```suggestion
Every OpenID4VP Authorization Request results in a response being provided through the W3C Digital Credentials API. The response is an instance of the `DigitalCredential` credential interface, as defined in [@!w3c.digital_credentials_api], and the OpenID4VP Authorization Response parameters as defined for the Response Type are represented as an object within the `data` attribute.
```

The following is a non-normative example of processing an unsigned OpenID4VP response that could be received from the W3C Digital Credentials API:

```js
const credential = await navigator.identity.get(request);
if (credential.protocol == "openid4vp") {
  // Extract relevant data members
  const { vp_token, presentation_submission } =  credential.data;
  // presentation_submission is a javascript object

>   // vp_token is a string or javascript object depending on the credential type

Reviewer 2: ```suggestion
  // vp_token is a string or javascript object depending on the credential type
}
```

```

The following is a non-normative example of processing an encrypted OpenID4VP response that could be received from the W3C Digital Credentials API:

```js
const credential = await navigator.identity.get(request);
if (credential.protocol == "openid4vp") {
  // Extract encrypted response
  const { response } =  credential.data;

>   // response is a string containing a JWE, now decrypt it

Reviewer 2: ```suggestion
  // response is a string containing a JWE, now decrypt it
}
```

## Response

> Every OpenID4VP Authorization Request MUST result in a response being provided through the W3C Digital Credentials API. The response is a JavaScript object, where the Authorization Response parameters as defined for the Response Type are encoded as members within the `data` member of this JavaScript object.

Reviewer: I guess it would make sense to link to the current draft here?

> Every OpenID4VP Authorization Request MUST result in a response being provided through the W3C Digital Credentials API. The response is a JavaScript object as defined in [@!w3c.digital_credentials_api], where the Authorization Response parameters as defined for the Response Type are encoded as members within the `data` member of this JavaScript object.

Reviewer 3: since the vp_token passes through the use-agent, it would be recommended to encrypt the JWT containing the response using the RP public key

Reviewer 4: there is a chance encryption will be mandatory in W3C spec, so i think we are waiting for the decision there?

Author: There's an issue that mentions encryption here:

https://github.com/openid/OpenID4VP/issues/203

We might need a more focussed issue on it, but I would like to keep it out of scope of this PR as we need to reach consensus in the working group(s) first.

> Every OpenID4VP Authorization Request MUST result in a response being provided through the W3C Digital Credentials API. The response is a JavaScript object as defined in [@!w3c.digital_credentials_api], where the Authorization Response parameters as defined for the Response Type are encoded as members within the `data` member of this JavaScript object.

Reviewer 4: ```suggestion
Every OpenID4VP Authorization Request MUST result in a response being provided through the W3C Digital Credentials API. This response is a JavaScript object as defined in [@!w3c.digital_credentials_api], where the OpenID4VP Authorization Response parameters as defined for the Response Type are encoded as members within the `data` member of this JavaScript object.
```

I would like to be very clear on browser API response vs openid4vp authorization response

> Every OpenID4VP Authorization Request MUST result in a response being provided through the W3C Digital Credentials API. This response is a JavaScript object as defined in [@!w3c.digital_credentials_api], where the OpenID4VP Authorization Response parameters as defined for the Response Type are encoded as members within the `data` member of this JavaScript object.

Reviewer 2: This should probably just be a statement of fact, as the W3C spec already normatively mandates that the response is always  a `DigitalCredential`: 

```suggestion
Every OpenID4VP Authorization Request results in a response being provided through the W3C Digital Credentials API. The response is an instance of a `DigitalCredential` credential interface, as defined in [@!w3c.digital_credentials_api], and the OpenID4VP Authorization Response parameters as defined for the Response Type are represented as objects within the `data` attribute.
```

Author: Thanks Marcos!

On this part of the change:

> the OpenID4VP Authorization Response parameters as defined for the Response Type are represented **as objects** within the `data` attribute

Is 'represented as objects' technically correct given the values might be strings?

Reviewer 2: Yeah, good point. What about if we only talk about the value of `.data`: "...Response Type is represented as an object within the `data` attribute" (updated above).

Author: Works for me - have changed an 'is' to 'are' as I think that reads better:

```suggestion
Every OpenID4VP Authorization Request results in a response being provided through the W3C Digital Credentials API. The response is an instance of a `DigitalCredential` credential interface, as defined in [@!w3c.digital_credentials_api], and the OpenID4VP Authorization Response parameters as defined for the Response Type are represented as an object within the `data` attribute.```

The following is a non-normative example of an unsigned OpenID4VP response that could be received from the W3C Digital Credentials API:

```js
{

>   protocol: "openid4vp",

Reviewer 2: A better way to represent this might be as follows, as the returned `DigitalCredential` object is not technically "js object":

```JS
const credential = await navigator.identity.get(request);
console.log(credential.protocol); // "openid4vp"

// Extract relevant data members
const { vp_token, presentation_submission } =  credential.data;

```

Author: Do you think it'd be over misleading to show it as an object?

If we go along the lines you suggest I'd probably prefer something a bit closer to how expect developers to use it, perhaps:

```
const credential = await navigator.identity.get(request);
if (credential.protocol == "openid4vp") {
  // Extract relevant data members
  const { vp_token, presentation_submission } =  credential.data;
  // presentation_submission is a javascript object
  // vp_token is a string or javascript object depending on the credential type
```

  data: {
    vp_token: "...",
    presentation_submission: {...}
  }
}
```

> The following is a non-normative example of an encrypted OpenID4VP response that could be received from the W3C Digital Credentials API:

Reviewer 2: Similarly.... 

```JS
const credential = await navigator.identity.get(request);
console.log(credential.protocol); // "openid4vp"

// Extract relevant data response
const { response } =  credential.data;
```
